api/tests/unit/environments/test_unit_environments_models.py

```diff
@@ -802,20 +804,85 @@ def test_get_hide_disabled_flags(
     assert environment.get_hide_disabled_flags() is expected_result
 
 
-def test_saving_environment_api_key_calls_put_item_with_correct_arguments_if_enabled(
-    dynamo_enabled_project_environment_one, mocker
+def test_saving_environment_api_key_creates_dynamo_document_if_enabled(
+    dynamo_enabled_project_environment_one: Environment,
+    mocker: MockerFixture,
+    flagsmith_environment_api_key_table: "Table",
 ):
     # Given
-    mocked_environment_api_key_wrapper = mocker.patch(
-        "environments.models.environment_api_key_wrapper", autospec=True
+    mocker.patch(
+        "environments.models.DynamoEnvironmentAPIKeyWrapper.table",
+        new_callable=mocker.PropertyMock,
+        return_value=flagsmith_environment_api_key_table,
     )
     # When
     api_key = EnvironmentAPIKey.objects.create(
         name="Some key", environment=dynamo_enabled_project_environment_one
     )
 
     # Then
-    mocked_environment_api_key_wrapper.write_api_key.assert_called_with(api_key)
+    response = flagsmith_environment_api_key_table.get_item(Key={"key": api_key.key})
+    assert response["Item"]["key"] == api_key.key
+
+
+def test_deleting_environment_api_key_deletes_dynamo_document_if_enabled(
+    dynamo_enabled_project_environment_one: Environment,
+    mocker: MockerFixture,
+    flagsmith_environment_api_key_table: "Table",
+):
+    # Given
+    mocker.patch(
+        "environments.models.DynamoEnvironmentAPIKeyWrapper.table",
+        new_callable=mocker.PropertyMock,
+        return_value=flagsmith_environment_api_key_table,
+    )
+    api_key = EnvironmentAPIKey.objects.create(
+        name="Some key", environment=dynamo_enabled_project_environment_one
+    )
+
+    # When
+    api_key.delete()
+
+    # Then
+    response = flagsmith_environment_api_key_table.get_item(Key={"key": api_key.key})
+    assert "Item" not in response
+
+
+def test_deleting_environment_creates_task_to_delete_dynamo_document_if_enabled(
+    dynamo_enabled_project_environment_one: Environment,
+    mocker: MockerFixture,
+) -> None:
+    # Given
+    mocked_task = mocker.patch("environments.tasks.delete_environment_from_dynamo")
+    mocker.patch(
+        "environments.models.environment_wrapper",
+        new_callable=mocker.PropertyMock,
+        return_value=True,
+    )
```

Comment:
I'm a little unsure what this is mocking? It seems to be trying to mock a property but I don't think it actually is?